docs/howtos/extending/magicgui.md

```diff
@@ -48,44 +89,89 @@ def widget_demo(
-purpose of this page is to document some of the conveniences provided by napari
-when using `magicgui` with napari-specific type annotations.
-
-## magicgui and type annotations
 
 `magicgui` uses [type hints](https://peps.python.org/pep-0484/) to infer
 the appropriate widget type for a given function parameter, and to indicate a
 context-dependent action for the object returned from the function (in the
-absence of a type hint, the type of the default value will be used).  Third
-party packages (like `napari` in this case) may provide support for their types
+absence of a type hint, the type of the default value will be used). You can also
+customize your widget using
+[`magicgui.magicgui`](https://pyapp-kit.github.io/magicgui/api/magicgui/#magicgui.magicgui)
+parameters. In the example above, `call_button` specifies the button text and the
+`param_options` `slider_float` and `dropdown` let you customize the widget
+associated with those function parameters.
+
+Third party packages (like `napari` in this case) may provide support for their types
 using
 [`magicgui.register_type`](https://pyapp-kit.github.io/magicgui/usage/types_widgets.html#register-type).
-This is how using the type annotations described below leads to widgets and/or
-"actions" in napari.
+Indeed `napari` uses
+[`magicgui.register_type`](https://pyapp-kit.github.io/magicgui/usage/types_widgets.html#register-type)
+to provide support for napari-specific type annotations. This makes it easy to
+use `magicgui` to build widgets in `napari`. Note all type annotations below *require*
+that the resulting widget be added to a napari viewer.
+
+Below we demonstrate how to create a simple threshold widget using `magicgui` and add
+it to the viewer. Note the `auto_call` parameter tells `magicgui` to call the function
+whenever a `threshold_magic_widget` parameter changes, thus the function is called
+as soon as we add the 'camera' image.
 
-```{important}
-All of the type annotations described below *require* that the resulting widget
-be added to a napari viewer (either via `viewer.window.add_dock_widget`, or
-by providing a magicgui-based widget via the {func}`~napari.plugins.hook_specifications.napari_experimental_provide_dock_widget` plugin hook specification).
+```{code-cell} python
+:tags: [remove-output]
+from skimage import data
+from skimage.util import img_as_float
+
+import napari
+
+@magicgui(
+    threshold={"widget_type": "FloatSlider", "max": 1}, auto_call=True
+)
+def threshold_magic_widget(
+    img_layer: "napari.layers.Image", threshold: "float"
+) -> "napari.types.LabelsData":
+    return img_as_float(img_layer.data) > threshold
+
+# Create the viewer and add an image
+viewer = napari.view_image(data.camera())
+# Add widget to viewer
+viewer.window.add_dock_widget(threshold_magic_widget)
 ```
 
+```{code-cell} python
+:tags: [remove-input]
+from napari.utils import nbscreenshot
+
+viewer.window._qt_window.resize(750, 550)
```

Comment:
could we maybe make this a bit bigger? The screenshots really squish up the image. If we can't I think it's nbd

Reply:
Done, hopefully better.